db/background_mgr.go

```diff
@@ -52,47 +76,160 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
+	Init(clusterStatus []byte) error
 	Run(options map[string]interface{}, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
 func (b *BackgroundManager) Start(options map[string]interface{}) error {
+	b.terminator = make(chan struct{}, 1)
+	b.terminatorClosed = false
+
 	err := b.markStart()
 	if err != nil {
 		return err
 	}
 
+	var processClusterStatus []byte
+	if b.clusterAwareOptions != nil {
+		processClusterStatus, _, err = b.clusterAwareOptions.bucket.GetRaw(b.clusterAwareOptions.StatusDocID())
+		if err != nil && !base.IsDocNotFoundError(err) {
+			return err
```

Comment:
If GetRaw returns an error, should we wrap the error to make it more meaningful to the caller of Start?  Want to make sure the error is meaningful in case it's getting passed all the way back to the REST API response.